### PR TITLE
CUPS plugin

### LINKED ISSUE #857: CUPS information
CUPS provides a lot of interesting statistics (jobs in queue, printer statuses, printing frequency), and can benefit from alerts (for example, if print jobs start piling up or one gets stuck).

It would be very nice to see a CUPS plugin for netdata.

### PR DESCRIPTION
**Pull request by simonnagl**
##### Summary

Adds a C plugin which collects metrics of the default cups daemon. See the new README.md for a list of charts provided.

##### Additional Information

Fixes #857
Replaces #1678

### CODE REVIEW

**Comment by vlvkobal** [comment]
@simonnagl It can't be compiled when CUPS is not installed. Also, please use
`CFLAGS="-O1 -ggdb -Wall -Wextra -Wformat-signedness -fstack-protector-all -DNETDATA_INTERNAL_CHECKS=1 -D_FORTIFY_SOURCE=2 -DNETDATA_VERIFY_LOCKS=1" ./netdata-installer.sh --disable-lto --dont-wait`
and eliminate all compiler warnings.

**Comment by simonnagl** [comment]
@vlvkobal Thank you for reviewing this. 

I fixed the issues with compiling the plugin with the additional CFLAGS you mentioned.
Also, I extended the check of cups.plugin being installed and now use cups-config to get the compiler flags. If there is something else to improve please tell me.

**Comment by vlvkobal** [comment]
One more warning:
```
In file included from collectors/cups.plugin/../../libnetdata/libnetdata.h:300,
                 from collectors/cups.plugin/cups_plugin.c:9:
collectors/cups.plugin/cups_plugin.c: In function 'main':
collectors/cups.plugin/cups_plugin.c:359:27: warning: format '%d' expects argument of type 'int', but argument 6 has type 'unsigned int' [-Wformat=]
                     error("Unsupported job state (%d) found.", curr_job->state);
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~
collectors/cups.plugin/../../libnetdata/log/log.h:85:81: note: in definition of macro 'error'
 #define error(args...)   error_int("ERROR", __FILE__, __FUNCTION__, __LINE__, ##args)
                                                                                 ^~~~
```

**Comment by simonnagl** [comment]
@vlvkobal Thank you! I did not get this warning, maybe because my compiler does not support `-Wformat-signedness`. 

I hope I could fix it with my latest commit. 

```
gcc --version

Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 10.0.0 (clang-1000.10.44.4)
Target: x86_64-apple-darwin18.2.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

File: collectors/cups.plugin/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+## Charts
+
+`cups.plugin` provides one common section `destinations` and one section per destination. 
+
+> Destinations in CUPS represent individual printers or classes (collections or pools) of printers (https://www.cups.org/doc/cupspm.html#working-with-destinations)
+
+The section `server` provides these charts:
+
+- destinations by state
+  - idle
```
Comment (simonnagl) [inline collectors/cups.plugin/README.md]: The documentation preview does not detect indented lists.

**Comment by simonnagl** [comment]
Open Task:

- [ ] Link cups.plugin in the documentation menu and add it to the external plugin table.

**Comment by simonnagl** [comment]
> Please add configuration instructions and any prerequisites to the README (doesn't need to be long).

@cakrit I added a short description. If you think more details like installation note etc. is required just tell me. Thank you for your review.

**Comment by cakrit** [comment]
> > Please add configuration instructions and any prerequisites to the README (doesn't need to be long).
> 
> @cakrit I added a short description. If you think more details like installation note etc. is required just tell me. Thank you for your review.

It's fine, thanks

File: collectors/cups.plugin/cups_plugin.c [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+
+    httpClose(http);
+    info("CUPS process exiting");
+}
+
+#else // !HAVE_CUPS
+
+int main(int argc, char **argv)
+{
+    fatal("freeipmi.plugin is not compiled.");
```
Comment (vlvkobal) [inline collectors/cups.plugin/cups_plugin.c]: Please change the name of the plugin (freeipmi -> cups).

**Comment by vlvkobal** [comment]
Thank you @simonnagl !

**Pull request merged**